src/porepy/models/solution_strategy.py

```diff
@@ -414,34 +414,29 @@ def before_nonlinear_iteration(self) -> None:
         # self.nonlinear_discretizations, this will be a no-op.
         self.rediscretize()
 
-    def after_nonlinear_iteration(self, solution_vector: np.ndarray) -> None:
+    def after_nonlinear_iteration(self, nonlinear_increment: np.ndarray) -> None:
         """Method to be called after every non-linear iteration.
 
         Possible usage is to distribute information on the new trial state, visualize
         the current approximation etc.
 
         Parameters:
-            solution_vector: The new solution, as computed by the non-linear solver.
+            nonlinear_increment: The new solution, as computed by the non-linear solver.
 
         """
-        self._nonlinear_iteration += 1
         self.equation_system.shift_iterate_values()
         self.equation_system.set_variable_values(
-            values=solution_vector, additive=True, iterate_index=0
+            values=nonlinear_increment, additive=True, iterate_index=0
         )
+        self.nonlinear_solver_statistics.num_iteration += 1
 
-    def after_nonlinear_convergence(
-        self, solution: np.ndarray, errors: float, iteration_counter: int
-    ) -> None:
+    def after_nonlinear_convergence(self, solution: np.ndarray) -> None:
```

Comment:
Is there a historical or important reason why `solution` has been offered as part of the signature here? Otherwise, I would remove it.

Reply:
@keileg @IvarStefansson two more questions, for which you possibly have an answer.

Reply:
I guess the answer is yes, there is some historical reason. But I see no reason to keep it, as evident from the fact that it's fetched from storage in line 442.

Reply:
Agreed.

Reply:
Removed.